core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1109,6 +1110,29 @@ static RelDataType deriveTypeSplit(SqlOperatorBinding operatorBinding,
           .withOperandTypeInference(InferTypes.RETURN_TYPE)
           .withKind(SqlKind.CONCAT_WS_MSSQL);
 
+  /** The "CONCAT_WS(separator[, str | array(str)]+)" function in (SPARK).
+   *
+   * <p>Differs from {@link #CONCAT_WS} (MySQL, Postgres) in that it accepts
+   * only 1 argument, and except for first argument, the following argument support array types.
```

Comment:
The Postgres support arryas. For example:
`select concat_ws(',',ARRAY[10000, 10000, 10000, 10000]);`

Reply:
1. Error comments have been removed.
2. Currently, Postgres `concat_ws` does not support arrays in the implementation of calcite, but it can be expanded in another JIRA.

Reply:
I log this in new issue https://issues.apache.org/jira/browse/CALCITE-6450